ophyd/signal.py

```diff
@@ -810,7 +921,116 @@ def limits(self):
         return (self._metadata['lower_ctrl_limit'],
                 self._metadata['upper_ctrl_limit'])
 
-    def get(self, *, as_string=None, connection_timeout=1.0, form='time',
+    def _get_with_retries(self, pv, timeout, connection_timeout, retries,
+                          as_string, form):
+        """
+        Utility method implementing a retry loop for get and get_setpoint
+
+        Returns info from pv.read_with_metadata(...) or raises TimeoutError
+        """
+        # Fall back to instance default if no value is given.
+        if timeout is DEFAULT_TIMEOUT:
+            timeout = self.timeout
+        if connection_timeout is DEFAULT_CONNECTION_TIMEOUT:
+            connection_timeout = self.connection_timeout
+        if retries is DEFAULT_READ_RETRIES:
+            retries = self.read_retries
+        retries = int(retries)  # normalize/validate input
+        floor = self.__read_attempt_timeout_floor  # alias for brevity
+        attempts = 1 + retries
+        start_time = time.monotonic()
+        if timeout is None:
+            # The user has told us to wait forever. How shall we divide
```

Comment:
We still don't actually wait forever as the number of retries can not be infinite (yet).